src/core_modules/capture-core/reducers/descriptions/workingLists/workingLists.reducerDescription.js

```diff
@@ -276,6 +276,33 @@ export const workingListsTemplatesDesc = createReducerDescription({
             },
         };
     },
+    [workingListsCommonActionTypes.TEMPLATE_SHARING_SETTINGS_SET]: (
+        state, {
+            payload: {
+                sharingSettings,
+                templateId,
+                storeId,
+            },
+        }) => {
+        const initialTemplates = state[storeId].templates;
+        const templateIndex = initialTemplates.findIndex(template => template.id === templateId);
+
+        const template = {
+            ...initialTemplates[templateIndex],
+            ...sharingSettings,
+        };
+
+        const updatedTemplates = [...initialTemplates];
+        updatedTemplates[templateIndex] = template;
```

Comment:
Looks like to me there is some business logic in this bit. Even though we dont have a hard rule on this, I think keeping the reducers slim and only for updating values  would make your solution more aligned with the rest of our reducers. 

This isnt a blocker ofc

Reply:
Totally agree. I need to refactor how the templates are stored, should be stored as an object literal instead of an array, this would fix the problem I've run into here.

This would require some effort to fix so I would rather do this in a separate PR though.